_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,983 @@
+        "_score": 0.015748845,
+        "_source": {
+          "text": "A wild animal races across an uncut field with a minimal amount of trees.",
+          "id": "1775029934.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "5",
+        "_score": 0.015177963,
+        "_source": {
+          "text": "A rodeo cowboy , wearing a cowboy hat , is being thrown off of a wild white horse .",
+          "id": "2691147709.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "1",
+        "_score": 0.013272902,
+        "_source": {
+          "text": "A West Virginia university women 's basketball team , officials , and a small gathering of fans are in a West Virginia arena .",
+          "id": "4319130149.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "3",
+        "_score": 0.011347735,
+        "_source": {
+          "text": "People line the stands which advertise Freemont 's orthopedics , a cowboy rides a light brown bucking bronco .",
+          "id": "2664027527.jpg"
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+### Search using a combined keyword search and neural search
+
+To combine keyword search and neural search, you need to set up a [search pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/index/) that runs at search time. The search pipeline you'll configure intercepts search results at an intermediate stage and applies the [`normalization_processor`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/) to them. The `normalization_processor` normalizes and combines the document scores from multiple query clauses, rescoring the documents according to the chosen normalization and combination techniques. 
+
+#### Step 1: Configure a search pipeline
+
+To configure a search pipeline with a `normalization_processor`, use the following request. The normalization technique in the processor is set to `min_max`, and the combination technique is set to `arithmetic_mean`. The `weights` array specifies the weights assigned to each query clause as decimal percentages:
+
+```json
+PUT /_search/pipeline/nlp-search-pipeline
+{
+  "description": "Post processor for hybrid search",
+  "phase_results_processors": [
+    {
+      "normalization-processor": {
+        "normalization": {
+          "technique": "min_max"
+        },
+        "combination": {
+          "technique": "arithmetic_mean",
+          "parameters": {
+            "weights": [
+              0.3,
+              0.7
+            ]
+          }
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Step 2: Search with the hybrid query
+
+You'll use the [`hybrid` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/hybrid/) to combine the `match` and `neural` query clauses. Make sure to apply the previously created `nlp-search-pipeline` to the request in the query parameter:
+
+```json
```

Comment:
Hey should we also pitch search templates here so that we dont have to worry about having the model id checked into code?

Reply:
Sure, let me add it as an "advanced" note.